File: saveTiffPNG.ijm.ijm.java
#run("Save");
#run ("Jpeg...");
from ij import IJ 
from ij.plugin.frame import RoiManager;
from ij.io import SaveDialog
import os
imp = IJ.getImage(); 
#sd = SaveDialog("Specify", "", ".roi");
#IJ.saveAs(imp, "Selection", sd.getDirectory() + sd.getFileName() + ".roi"); 

File: saveTiffPNG.ijm.ijm.py
#run("Save");
#run ("Jpeg...");
from ij.io import *;
from ij.plugin.frame import *;
from ij import IJ;

rm = RoiManager.getInstance()
imp = IJ.getImage(); 
roi = imp.getRoi();
sd = SaveDialog("specify", IJ.getDirectory("image"), ".roi");
RoiEncoder.save(roi, sd.getDirectory() + sd.getFileName()) ;
rm.addRoi(roi);
index = rm.getCount()-1;
rm.rename(index, sd.getFileName());